src/transform/src/lib.rs

```diff
@@ -329,6 +330,12 @@ where
         }
         Ok(result) => result.map_err(|e| e),
         Err(panic) => {
+            // A panic during optimization is always a bug; log an error so we learn about it.
+            tracing::error!(
+                "caught a panic during query optimization: {panic}\n{}",
+                Backtrace::force_capture()
```

Comment:
(This `Backtrace::force_capture` doesn't capture the backtrace of the panic, but just the backtrace of where we catch, i.e., of `catch_unwind_optimize`, right? This can also be somewhat useful, but it would be much more useful to capture the backtrace of the panic itself somehow. But it's ok if this is out of scope for this PR.)

Reply:
Ah, good catch! I think it's actually better to remove the backtrace printing here, because otherwise it could lead to confusion. People could assume that the backtrace was from the place that caused the panic, instead of the place that caught it.

I think to get the actual backtrace we'd have to collect it in the enhanced panic handler, and then smuggle it through to `catch_unwind` somehow. Probably through static variables like the `CATCHING_UNWIND` we already use. I'll leave that as a follow up.